### PR TITLE
test/rgw: fixes for test_multi_period_incremental_sync()

### PR DESCRIPTION
**Pull request by cbodley**
test was only creating objects in subsequent periods, which wasn't adding any entries to the mdlog. this wasn't correctly testing incremental metadata sync across periods

### CODE REVIEW

File: src/test/rgw/test_multi.py [hunk 1]
```diff
@@ -846,39 +846,40 @@ def test_multi_period_incremental_sync():
...
     # wait for zone 1 to sync
     realm.zone_meta_checkpoint(realm.get_zone('us-1'))
 
     # change master back to zone 1 -> period 3
     realm.set_master_zone(realm.get_zone('us-1'))
 
-    for zone, bucket_name in zone_bucket.items():
+    # create another bucket in each zone, except for z3
+    for zone in realm.get_zones():
         if zone == z3:
```
Comment (yehudasa) [inline src/test/rgw/test_multi.py]: @cbodley are we assuming this test runs with at least 3 zones?

Comment (cbodley) [inline src/test/rgw/test_multi.py]: yes, the test is skipped if there are less than 3:
```python
def test_multi_period_incremental_sync():
    if len(realm.clusters) < 3:
        from nose.plugins.skip import SkipTest
        raise SkipTest("test_multi_period_incremental_sync skipped. Requires 3 or more clusters.")
```

**Comment by cbodley** [comment]
added a `time.sleep(5)` to `set_master_zone()` to resolve a race between period commit and later metadata operations

**Comment by cbodley** [comment]
@yehudasa thoughts on merging this before the fixes in https://github.com/ceph/ceph/pull/13070? the mdlog trimming pr at https://github.com/ceph/ceph/pull/13111 has changes to `test_multi.py` that depend on this

**Comment by cbodley** [comment]
rebased over multi-zonegroup changes. still passing against https://github.com/ceph/ceph/pull/13070

**Comment by yehudasa** [comment]
@cbodley feel free to merge at your convenience

**Comment by cbodley** [comment]
rebased, will merge on build check

**Comment by cbodley** [comment]
jenkins test this please

**Pull request merged**